src/server/pfs/server/driver.go

```diff
@@ -325,18 +345,31 @@ func (d *driver) deleteRepo(txnCtx *txnenv.TransactionContext, repo *pfs.Repo, f
 		return err
 	}
 
-	repoInfo := new(pfs.RepoInfo)
-	if err := repos.Get(repo.Name, repoInfo); err != nil {
-		if !col.IsErrNotFound(err) {
-			return errors.Wrapf(err, "repos.Get")
+	// if this is a user repo, delete any dependent repos
+	if repo.Type == pfs.UserRepoType {
```

Comment:
Where is the current design doc for System Repos? After the last meeting we had about this, I was under the impression that this operation could be done without a read.

Reply:
I'm mostly curious about the table structure here.

Reply:
doc is [here](https://docs.google.com/document/d/1eq4_wIbh7-M4r3hcZWGFEPYU9FN8eGQIp0b4dQaORxA/edit#heading=h.3ivaro2ikgbw), and [this](https://github.com/pachyderm/pachyderm/pull/6070/files#diff-9c8c0ec115494d42aeabac6cddc20e58b1bc9fd5d44f05b879007998d08de8fb) is the specific PR.

I'm not exactly sure what you mean by "this operation", we can definitely delete the repo entries directly, but I don't see how the other components of `DeleteRepo` could happen as easily.